packages/odyssey-react-mui/src/labs/DataTable.tsx

```diff
@@ -383,6 +389,132 @@ const DataTable = ({
   const rowVirtualizerInstanceRef =
     useRef<MRT_Virtualizer<HTMLDivElement, HTMLTableRowElement>>(null);
 
+  const getRowFromTableAndSetHovered = (
+    table: TableType,
+    id: MRT_RowData["id"]
+  ) => {
+    const nextRow: MRT_RowData = table.getRow(id);
+
+    if (nextRow) {
+      table.setHoveredRow(nextRow);
+    }
+  };
+
+  const resetDraggingAndHoveredRow = (table: TableType) => {
+    setDraggingRow(null);
+    table.setHoveredRow(null);
+  };
+
+  type HandleDragHandleKeyDownArgs = {
+    table: TableType;
+    row: MRT_Row<MRT_RowData>;
+    event: KeyboardEvent<HTMLButtonElement>;
+  };
+
+  const handleDragHandleKeyDown = useCallback(
+    ({ table, row, event }: HandleDragHandleKeyDownArgs) => {
+      const { hoveredRow } = table.getState();
+
+      const { key } = event;
+
+      const isSpaceKey = key === " ";
+      const isEnterKey = key === "Enter";
+      const isEscapeKey = key === "Escape";
+      const isArrowDown = key === "ArrowDown";
+      const isArrowUp = key === "ArrowUp";
+      const isSpaceOrEnter = isSpaceKey || isEnterKey;
+      const zeroIndexedPageNumber = page - 1;
+      const currentIndex = row.index + zeroIndexedPageNumber * resultsPerPage;
+
+      if (isEscapeKey) {
+        resetDraggingAndHoveredRow(table);
+        return;
+      }
+
+      if (isSpaceOrEnter) {
+        event.preventDefault();
+        event.stopPropagation();
+      }
+
+      if (draggingRow) {
+        if (typeof hoveredRow?.index === "number") {
+          const { index } = hoveredRow;
+
+          if (isSpaceOrEnter) {
+            const indexByPage =
+              zeroIndexedPageNumber > 0
+                ? index + zeroIndexedPageNumber * resultsPerPage
+                : index;
+
+            if (indexByPage !== currentIndex) {
+              handleRowReordering({
+                rowId: row.id,
+                newIndex: indexByPage,
+              });
+
+              // Can't transition CSS hover effect. Use timeout to delay hovered row effect removal
+              setTimeout(() => {
+                resetDraggingAndHoveredRow(table);
+              }, odysseyDesignTokens.TransitionDurationMainAsNumber);
+              return;
+            }
+          }
+
+          if (isArrowDown || isArrowUp) {
+            const nextIndex = isArrowDown ? index + 1 : index - 1;
+            const nextRowFromData = data[nextIndex];
+
+            if (nextRowFromData) {
+              getRowFromTableAndSetHovered(table, nextRowFromData.id);
+            }
+          }
+        } else {
+          if (isArrowDown || isArrowUp) {
+            const nextIndex = isArrowDown ? row.index + 1 : row.index - 1;
+
+            const nextRowFromData = data[nextIndex];
+
+            if (nextRowFromData) {
+              getRowFromTableAndSetHovered(table, nextRowFromData.id);
+            }
```

Comment:
Does this work?

```suggestion
          if (isArrowDown || isArrowUp) {
            const nextIndex = isArrowDown ? index + 1 : index - 1;

              getRowFromTableAndSetHovered(table, data[nextIndex]?.id);
          }
        } else {
          if (isArrowDown || isArrowUp) {
            const nextIndex = isArrowDown ? row.index + 1 : row.index - 1;

          getRowFromTableAndSetHovered(table, data[nextIndex]?.id);
```

I'm trying to simplify the logic even more. I personally like to remove `if` checks and the creation of new variables where I can as they can make the code harder to read in some cases.

Reply:
@chrispulsinelli-okta Had suggested to abstract this logic out. Could be more readable once I've moved this arrow press handling into a separate function